src/components/web-monitoring-ui.tsx

```diff
@@ -57,3 +71,18 @@ export default class WebMonitoringUi extends React.Component<undefined, IWebMoni
         return {api};
     }
 }
+
+function getPagesByUser (userName: string): Promise<any> {
+    const url = `${window.location.origin}/domains/${userName}`;
+    return fetch(url)
+        .then(response => response.json())
+        .then(data => {
+            if (data.domains) {
+                return api.getPagesByDomains(data.domains).then(pages => pages);
+            } else {
+                return Promise.reject(data.error);
+            }
+        })
+        .then(pages => pages)
+        .catch(err => Promise.reject(err));
```

Comment:
Yep. I must have been doing things weird when I first started because I couldn't get the list view to render before without the extra `then`. But tested and looks like everything is fine now lol. Will remove.